### PR TITLE
[DO NOT MERGE][AKS] Add new api model

### PR DESCRIPTION
**Pull request by zqingqing1**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (dsgouda) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Please use `x-ms-enum` instead

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (dsgouda) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Please check these properties for `readonly` ness

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: The value should be passed by customer, and default value is nil. so we are good here

**Comment by zqingqing1** [comment]
Please don't merge before I do some test. Will let you final review when this PR is ready.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (JackQuincy) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: We have actually stopped reading this from the customer. We just default to always use VMSS for this api version, on creates, old clusters we treat as not changing from what they are. So we should be able to remove this from the schema.

**Comment by dsgouda** [comment]
Hi @zqingqing1 will close this PR by EOW if there is no activity.

**Comment by zqingqing1** [comment]
Please don't close this PR, There is still some work from Laurent to support multi-api, after then, This PR should be good to merge.

**Comment by dsgouda** [comment]
@zqingqing1 You can always open a new PR when you are ready to go. We do not recommend keeping PRs open against the repo for too long.

**Pull request closed without merging**
